staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4981,6 +4981,20 @@ type ServiceProxyOptions struct {
 }
 
 // ObjectReference contains enough information to let you inspect or modify the referred object.
+// ---
+// New uses of this type are discouraged because of difficulty describing its usage when embedded in APIs.
+//  1. Ignored fields.  It includes many fields which are not generally honored.  For instance, ResourceVersion and FieldPath are both very rarely valid in actual usage.
+//  2. Invalid usage help.  It is impossible to add specific help for individual usage.  In most embedded usages, there are particular
+//     restrictions like, "must refer only to types A and B" or "UID no honored" or "name must be restricted".
+//     Those cannot be well described when embedded.
+//  3. Inconsistent validation.  Because the usages are different, the validation rules are different by usage, which makes it hard for users to predict what will happen.
+//  4. The fields are both imprecise and overly precise.  Kind is not a precise mapping to a URL. This can produce ambiguity
+//     during interpretation and require a REST mapping.  In most cases, the dependency is on the group,resource tuple
+//     and the version of the actual struct is irrelevant.
+//  5. We cannot change it (or will we...).  Because this type is embedded in many locations, updates to this type
+//     will affect numerous schemas.  Don't make new APIs which interpret references beholden to an API they do not control.
```

Comment:
I'm having trouble parsing this... would this capture what you meant:
```suggestion
//     will affect numerous schemas.  Don't make new APIs embed an underspecified API type they do not control.
```